pkgs/applications/editors/jupyter-kernels/clojupyter/default.nix

```diff
@@ -0,0 +1,73 @@
+{ pkgs
+, stdenv
+, lib
+, jre
+, fetchFromGitHub
+, writeShellScript
+, runCommand
+, imagemagick
+}:
+
+# To test:
+# $(nix-build -E 'with import <nixpkgs> {}; jupyter.override { definitions = { clojure = clojupyter.definition; }; }')/bin/jupyter-notebook
+
+let
+  cljdeps = import ./deps.nix { inherit pkgs; };
+  classp  = cljdeps.makeClasspaths {};
+
+  shellScript = writeShellScript "clojupyter" ''
+    ${jre}/bin/java -cp ${classp} clojupyter.kernel.core $*
```

Comment:
Could we get rid of that wrapper, by just inlining the java invokation into the definition argv? Doesn't seem like this is used elsewhere ...

Reply:
We definitely want to expose the `clojupyter` executable itself so it can be used in other ways. You can see the same approach is taken in the Octave kernel. I created that wrapper separately from the exposed `launcher` derivation because I wanted to attach the `name`, `pname`, and `meta` to the latter, and it seems impossible to do that with `writeShellScript` alone...

Reply:
maybe an `exec`, then ...

Reply:
.. also, I think the ususal sh idiom, to make sure that quoted arguments get passed correctly, is `"$@"` instead of `$*`

Reply:
Good point, fixed the arguments to be `"$@"`.

I don't think using `exec` will allow us to get rid of the wrapper (since `exec` is a shell primitive, and afaik the argv is not guaranteed to run in a shell) so I'm inclined to leave it as-is.

Reply:
Maybe I'm reading this wrong? https://man7.org/linux/man-pages/man1/exec.1p.html

```
If exec is specified with command, it shall replace the shell
with command without creating a new process.
```

Reply:
FWIW, it's used in makeWrapper https://github.com/NixOS/nixpkgs/blob/5fc7992364b278e2aa93914a6adccffefcf2ed67/pkgs/build-support/setup-hooks/make-wrapper.sh#L109 as well as many nix packages https://github.com/NixOS/nixpkgs/search?l=Shell&p=3&q=exec

Also note that it couldn't be a separate binary by its very nature, and it's in every shell since bourne shell .. https://www.cs.ait.ac.th/~on/O/oreilly/unix/upt/ch45_07.htm

But it's also a tiny optimization, probably not even worth the original feedback.